lib/bamboo/template.ex

```diff
@@ -0,0 +1,271 @@
+defmodule Bamboo.Template do
+  @moduledoc """
+  Render emails with layouts, view modules, and templates.
+
+  This module allows rendering emails with layouts and views. Pass an
+  atom (e.g. `:welcome_email`) as the template name to render both HTML and
+  plain text emails. Use a string if you only want to render one type, e.g.
+  `"welcome_email.text"` or `"welcome_email.html"`.
+
+  ## Examples
+
+  _Set the text and HTML layout for an email._
+
+      defmodule MyApp.Email do
+        use Bamboo.Template, view: MyApp.Email.AccountView
+
+        def welcome_email do
+          new_email()
+          |> put_text_layout({MyApp.Email.LayoutView, "email.text"})
+          |> put_html_layout({MyApp.Email.LayoutView, "email.html"})
+          |> render(:welcome) # Pass atom to render html AND plain text templates
+        end
+      end
+
+  _Set both the text and HTML layout at the same time for an email._
+
+      defmodule MyApp.Email do
+        use Bamboo.Template, view: MyApp.Email.AccountView
+
+        def welcome_email do
+          new_email()
+          |> put_layout({MyApp.Email.LayoutView, :email})
+          |> render(:welcome)
+        end
+      end
+
+  _Render both text and html emails without layouts._
+
+      defmodule MyApp.Email do
+        use Bamboo.Template, view: MyApp.Email.AccountView
+
+        def welcome_email do
+          new_email()
+          |> render(:welcome)
```

Comment:
```suggestion
          render(new_email(), :welcome)
```

Just to conform to community style conventions.

Reply:
I actually like the way it is for a few reasons: 

1. Since it's documentation, it's easier for people to read each step separately. The function call within a function call looks less clear at a glance. If it were something like `render(%Email{}, :welcome)` then maybe I'd consider it more. 
2. These functions would typically be used in a longer pipeline, so (I think) there is an _expectation_ of piping to `render/3` in people's minds. I'm not sure I'm explaining this correctly, but there are some functions that are _meant_ to be used with a pipe, regardless of how many arguments have gone before it. They _read_ better when piped into. So our brains are used to thinking of that function as something that gets piped into it (e.g. Wallaby's `assert_has` reads much better piped into `session |> assert_has(text("hello world"))` vs `assert_has(session, text("hello world"))`. Of course, there are functions that are the opposite way (like Phoenix's `link` for example. That would look bad piped into `"hello" |> link(to: Routes....)`)
3. Since this is typically used in pipelines, I think it makes it easier for people to copy into their own code. 

What do you think?